### ISSUE TITLE
determinePreferredLanguage() + registerAvailableLanguageKeys() still try to load unavailable language

### ISSUE THREAD

**Issue by PaulAnnekov** [angular-translate/angular-translate#1390]
When using `determinePreferredLanguage()` function to automatically detect language based on Navigator and `registerAvailableLanguageKeys()` to set available languages list, if unavailable language detected, then angular translate still tries to load it using translation loader. Why does it do this? It's unavailable.
POC: [plunker](https://plnkr.co/edit/cm18nwAYmVcohYtWNDSe?p=preview)

**Comment by knalli** [comment]
Yeah, this sounds valid.

We have to revisit https://github.com/angular-translate/angular-translate/blob/c5c20ac7e06d8569c79019b340102ce30fad8572/src/service/translate.js#L183-L185

**Comment by tspaeth** [comment]
Question - Implementation should be:

No matter, what determinePreferredLanguage analyzed - the registerAvailableLanguageKeys - output defines the valid languages. 
Therefore if your browser is de-DE and this is the preferred due to this - which of the languages shall be chosen as the active one by the module?

Usually, you could now define this wildcard mapping as a second parameter (maybe needing the correct mappings in before - didn'T test it:
```
registerAvailableLanguageKeys(languages, {
      "*": "ar"
    })
```

**Comment by knalli** [comment]
> Therefore if your browser is de-DE and this is the preferred due to this - which of the languages shall be chosen as the active one by the module?

The fallback one.

Perhaps we should log a warning if the combo preferredLanguage + available + fallback will result into nothing..

**Comment by knalli** [comment]
Fixed with bde935e -- with test case.
